Shoko.Server/Services/AbstractUserDataService.cs

```diff
@@ -235,6 +240,80 @@ public async Task<bool> SetEpisodeWatchedStatus(IShokoUser user, IShokoEpisode e
 
     #endregion
 
+    #region Series User Data
+
+    public async Task VoteOnSeries(IShokoSeries series, decimal voteValue, VoteType voteType, IShokoUser? user = null)
+    {
+        ArgumentNullException.ThrowIfNull(series);
+
+        if (user == null)
+            user = RepoFactory.JMMUser.GetAll().FirstOrDefault(u => u.IsAdmin == 1);
+
+        ArgumentNullException.ThrowIfNull(user);
+
+        if (series is not SVR_AnimeSeries svrSeries)
+            throw new ArgumentException("Series must be a SVR_AnimeSeries", nameof(series));
+
+        if (svrSeries.AniDB_Anime is not { } anidbAnime)
+            throw new ArgumentException("AniDB anime is not available for the series! Aborting!");
+
+        if (voteValue < -1 || voteValue > 10)
+            throw new ArgumentOutOfRangeException(nameof(voteValue), "Vote value must be between -1 and 10");
+
+        var anidbVoteType = voteType == VoteType.Permanent ? AniDBVoteType.Anime : AniDBVoteType.AnimeTemp;
+
+        // Handle deletion case
+        if (voteValue == -1)
```

Comment:
or we can convert this check to handle everything below zero (`voteValue < 0`) if we don't want to do the changes to the range check above.